fdbserver/ConfigBroadcastInterface.h

```diff
@@ -0,0 +1,138 @@
+	static constexpr FileIdentifier file_identifier = 9854021;
+
+	bool operator==(ConfigClassSet const& rhs) const { return classes == rhs.classes; }
+	bool operator!=(ConfigClassSet const& rhs) const { return !(*this == rhs); }
+
+	ConfigClassSet() = default;
+	ConfigClassSet(VectorRef<KeyRef> configClasses) {
+		for (const auto& configClass : configClasses) {
+			classes.insert(configClass);
+		}
+	}
+	ConfigClassSet(std::vector<KeyRef> configClasses) {
+		for (const auto& configClass : configClasses) {
+			classes.insert(configClass);
+		}
+	}
+
+	bool contains(KeyRef configClass) const { return classes.count(configClass); }
+	std::set<Key> const& getClasses() const { return classes; }
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, classes);
+	}
+};
+
+template <>
+struct Traceable<ConfigClassSet> : std::true_type {
+	static std::string toString(ConfigClassSet const& value) { return describe(value.getClasses()); }
+};
+
+struct ConfigBroadcastSnapshotReply {
+	static constexpr FileIdentifier file_identifier = 8701984;
+
+	ConfigBroadcastSnapshotReply() = default;
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar);
+	}
+};
+
+struct ConfigBroadcastSnapshotRequest {
+	static constexpr FileIdentifier file_identifier = 10911925;
+	Version version{ 0 };
+	std::map<ConfigKey, KnobValue> snapshot;
+	ReplyPromise<ConfigBroadcastSnapshotReply> reply;
+
+	ConfigBroadcastSnapshotRequest() = default;
+	template <class Snapshot>
+	explicit ConfigBroadcastSnapshotRequest(Version version, Snapshot&& snapshot)
+	  : version(version), snapshot(std::forward<Snapshot>(snapshot)) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, version, snapshot, reply);
+	}
+};
+
+struct ConfigBroadcastChangesReply {
+	static constexpr FileIdentifier file_identifier = 4014928;
+
+	ConfigBroadcastChangesReply() = default;
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar);
+	}
+};
+
+struct ConfigBroadcastChangesRequest {
+	static constexpr FileIdentifier file_identifier = 601281;
+	Version mostRecentVersion;
+	Standalone<VectorRef<VersionedConfigMutationRef>> changes;
+	ReplyPromise<ConfigBroadcastChangesReply> reply;
+
+	ConfigBroadcastChangesRequest() = default;
+	explicit ConfigBroadcastChangesRequest(Version mostRecentVersion,
+	                                       Standalone<VectorRef<VersionedConfigMutationRef>> const& changes)
+	  : mostRecentVersion(mostRecentVersion), changes(changes) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, mostRecentVersion, changes, reply);
+	}
+};
+
+/*
+ * The ConfigBroadcaster uses a ConfigBroadcastInterface from each worker to
+ * push updates made to the configuration database to the worker.
+ */
+class ConfigBroadcastInterface {
+	UID _id;
+
+public:
+	static constexpr FileIdentifier file_identifier = 1676543;
+	RequestStream<ConfigBroadcastSnapshotRequest> getSnapshot;
+	RequestStream<ConfigBroadcastChangesRequest> getChanges;
```

Comment:
These promise streams should be renamed now that the message types have been renamed.